datafusion/core/tests/sql/expr.rs

```diff
@@ -917,19 +917,19 @@ async fn test_interval_expressions() -> Result<()> {
     );
     test_expression!(
         "interval '0.499 day'",
-        "0 years 0 mons 0 days 11 hours 58 mins 33.596 secs"
+        "0 years 0 mons 0 days 11 hours 58 mins 33.600 secs"
     );
     test_expression!(
         "interval '0.4999 day'",
-        "0 years 0 mons 0 days 11 hours 59 mins 51.364 secs"
+        "0 years 0 mons 0 days 11 hours 59 mins 51.360 secs"
     );
     test_expression!(
         "interval '0.49999 day'",
         "0 years 0 mons 0 days 11 hours 59 mins 59.136 secs"
     );
     test_expression!(
         "interval '0.49999999999 day'",
-        "0 years 0 mons 0 days 12 hours 0 mins 0.00 secs"
+        "0 years 0 mons 0 days 11 hours 59 mins 59.999999136 secs"
     );
     test_expression!(
```

Comment:
i think we can add a test case for this (which will fail in the current version) and wait for the update to arrow 27 #4199
```suggestion
    test_expression!(
        "interval '0.00000000001 day'",
        "0 years 0 mons 0 days 0 hours 0 mins 0.000000864 secs"
    );
    test_expression!(
```

Reply:
i just found that the arrow-rs fix isn't included in 27.0.0
@tustvold do you have any suggestion for this? should we just merge this first and add some test cases after arrow 28.0.0 merged?

Reply:
We can also add test cases that are marked as `#[should_fail]`